src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.Configuration.targets

```diff
@@ -10,25 +10,33 @@ Copyright (c) .NET Foundation. All rights reserved.
 ***********************************************************************************************
 -->
 <Project ToolsVersion="14.0">
+
+  <PropertyGroup>
+    <_TargetingNETCoreApp30OrLater Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(_TargetFrameworkVersionWithoutV)' &gt; '2.9'">true</_TargetingNETCoreApp30OrLater>
+  </PropertyGroup>
+  
   <!--
       Initialize properties and items inferred using the project version. This file is not imported in projects referencing
       MVC \ Razor 2.2 or earlier since values specified here are provided via targets and props imported from NuGet packages
       such as Microsoft.AspNetCore.Razor.Design.
 
       The properties and items here are designed to be read by CPS so they should be just simple evaluation-time values
       and should not require targets to initialize.
+
+      Also, these values are based on whether the library is targeting netcoreapp3.0 for now. A better heuristic
+      would be to detect whether the ASP.NET Core shared framework is referenced.
   -->
   <PropertyGroup>
     <!--
       MVC uses a ProvideApplicationPartFactoryAttribute on the generated assembly to load compiled views from assembly. Set this to false, to prevent generating this attribute.
     -->
-    <GenerateProvideApplicationPartFactoryAttribute Condition="'$(GenerateProvideApplicationPartFactoryAttribute)'==''">true</GenerateProvideApplicationPartFactoryAttribute>
```

Comment:
This wasn't a safe assumption. We want to use this for netstandard2.0 class libraries that build components, but don't reference MVC or Razor.Hosting.

Reply:
Note, that I didn't really fix the underlying issue - these features should probably be opt-in.